### PR TITLE
feat(react): implement `useBlockProfiles` & `useUnblockProfiles`

### PR DESCRIPTION
**Pull request by reecejohnson**

### CODE REVIEW

File: examples/web/src/profiles/ProfilesPage.tsx [hunk 1]
```diff
@@ -3,81 +3,91 @@ import { LinkCard } from '../components/LinkCard';
...
   {
     label: 'useUpdateFollowPolicy',
-    description: `Update the follow policy for the logged-in Profile.`,
+    description: 'Update the follow policy for the logged-in Profile.',
     path: '/profiles/useUpdateFollowPolicy',
   },
+  {
+    label: 'useUnblockProfiles',
+    description: `Unblock profiles.`,
+    path: '/profiles/useUnblockProfiles',
```
Comment (krzysu) [inline examples/web/src/profiles/ProfilesPage.tsx:77]: do we need this separately from the 'useBlockProfiles & useUnblockProfiles' item below in this list?

File: examples/web/src/App.tsx [hunk 2]
```diff
@@ -144,7 +145,9 @@ export function App() {
                       <Route path="useProfileActionHistory" element={<UseProfileActionHistory />} />
                       <Route path="useSetProfileMetadata" element={<UseSetProfileMetadata />} />
                       <Route path="useUpdateFollowPolicy" element={<UseUpdateFollowPolicy />} />
+                      <Route path="useUnblockProfiles" element={<UseBlockAndUnblockProfiles />} />
```
Comment (krzysu) [inline examples/web/src/App.tsx:148]: do we need two routes to the same example?

File: packages/react/src/transactions/useSetProfileMetadata.ts [hunk 3]
```diff
@@ -87,12 +87,8 @@ export function useSetProfileMetadata(): UseDeferredTask<
...
   return useDeferredTask(async ({ metadataURI }: UseSetProfileMetadataArgs) => {
     invariant(
-      session?.authenticated,
-      'You must be authenticated to set profile metadata. Use `useLogin` hook to authenticate.',
-    );
-    invariant(
-      session.type === SessionType.WithProfile,
-      'You must have a profile to set profile metadata.',
+      session?.type === SessionType.WithProfile,
+      'You must be authenticated with a profile to block a profile. Use `useLogin` hook to authenticate.',
```
Comment (krzysu) [inline packages/react/src/transactions/useSetProfileMetadata.ts:91]: I think this message is in the wrong hook here, did you intend to change the logic in `useSetProfileMetadata` hook?

**Review by krzysu — APPROVED** [review]
just a few comments, the rest looks good! 👍

File: packages/react/src/transactions/useSetProfileMetadata.ts (hunk 3 above)
Comment (reecejohnson) [reply packages/react/src/transactions/useSetProfileMetadata.ts:91]: ah good spot, thanks!

**Pull request merged**